Automattic/wp-calypso
# Mastodon: swap feature flag for site feature check

Related to #76983 

## Proposed Changes

This PR removes the `mastodon` feature flag, and gates the Mastodon connection with a site feature check instead.

## Testing Instructions



### Prerequisites
- Make sure you have a Mastodon account
- Note the blog id of your test site
- Open the dev console

### Regression Testing
- Spin up Calypso by running this branch locally or by using a live link below
- Visit `/marketing/connections/:site`
- Make sure the section behaves as expected and that you don't see Mastodon as an eligible connection
- Check there's no error in the console

<img width="400" alt="Screenshot 2023-05-16 at 11 36 11 AM" src="https://github.com/Automattic/wp-calypso/assets/1620183/a51dd8e9-8eb7-4ef4-af65-c3b7a3cf0f1e">


### Testing
- Patch [this diff](D111014-code) and sandbox the API
- Repeat the steps above
- Add the blog sticker mentioned in the diff to your site
- Reload the page and notice the Mastodon item
- Test that you can connect/disconnect your site to Mastodon

<img width="400" alt="Screenshot 2023-05-16 at 11 36 22 AM" src="https://github.com/Automattic/wp-calypso/assets/1620183/eaf5f82c-43ab-4e60-8ece-92c96b851c37">


## Pre-merge Checklist
